engineering/methodology.md

```diff
@@ -1,131 +1,196 @@
+* Product does not need to be deployed, but the demo should be as real as possible.
+* Should be concise, respecting other people's time.
+* Fix max time **per-team: 15min (including questions)**. Agree between teams in advance, if more time needed.
 
-  + Deploying may not make sense for some teams, i. e. Data Science and Devops.
+## Keeping track of things
 
-  + Tools and Data Retrieval will deploy on Mondays, after the end of an sprint,
-    as they don't have the planing meeting until Tuesday.
+Github Projects board will used as a Kanban board.
 
-### Demo
+Teams managing issues across Github Org borders (Applications, Language Analysis) will use the “add card”  trick, described in dear-github/dear-github#209 to  add issues from different org  to the team’s board.
 
-Demo meetings
+Issues repositories:
 
-  + Idea: let know everybody what's going on (on quite technical level).
+* [src-d/issues-general](https://github.com/src-d/issues-general) - should be used more, as place for feedback/requests (i.e website, etc)
+* [src-d/issues-infrastructure](https://github.com/src-d/issues-infrastructure) - issues/requests to Infrastructure team must be created here.
 
-  + At the end of each sprint: Fridays 13:30.
+# Definitions
 
-  + Product does not need to be deployed, but the demo should be as real as
-    possible.
+Below the definitions of Teams Kanban columns are given.
 
-  + It's for eng team and stakeholders: POs, Mgt team (Jorge, Eiso, Marge, Maximo, Marcelo).
+This is not mere a formality - it’s important to have same things in mind when
+moving work from one state to another and what is documented below is a
+consensus, that took few hours to build among Engineering leads.
 
-  + Should be concise, respecting other people's time.
+## What does TODO mean?
 
-  + Fix max time per-team **total: 90min**, **per-team: 20min (15/5 questions)**.
+Something an engineer can immediately take and act up on. If the task is not
+possible to start until another (well-known) task is finished first, this task
+shall not be in TODO yet.
 
-  +  Agree between teams in advance, if more time needed.
+*Rationale* there are 2 goals we want to keep in mind, while working with Kanban
+board:
 
+* as small TODO list on a Kanban as possible and
+* tasks, moving fast from end-to-end of kanban board (short Cycle Time)
 
-### Retrospectives
+This definition and goals also imply that leads should grooming backlog on the
+daily basis.
 
-  + Starting at the end of the second sprint, one every other sprint.
+> Alternative definition (discussed but discarded): all things that are planned to get done ASAP. This would imply that issues can be added to TODO in a bulk at once, even in case when multiple issues of single project have known dependency on one another. Was discarded on the premise that it would go against the 2 goals, listed above.
```

Comment:
If that was discarded then this probably should be removed from the document.

Reply:
Inclusion of this option was in particular requested by @mcuadros as this workflow will be improved and thus it's nice to not loose the context with other possible options